plutus-use-cases/src/Language/Marlowe/Compiler.lhs

```diff
@@ -0,0 +1,841 @@
+
+This implies that \emph{Remaining Contract} and its \emph{State} are publicly visible.
+
+\subsection{Null}
+Possibly allow redeem of cash spent by mistake on this address? How?
+
+If we have all chain of txs of a contract we could allow redeems of mistakenly put money,
+and that would allow a contract creator to withdraw the contract initialization payment. \ref{ContractInit}
+
+
+\subsection{CommitCash}
+
+Alice has 1000 Ada in AliceUTXO.
+
+\begin{tikzpicture}[
+squarednode/.style={rectangle, draw=orange!60, fill=orange!5, very thick, minimum size=10mm},
+]
+%Nodes
+\node[squarednode,align=center] (contract) {Contract\\redeemer = CC #1 v:100 t:256};
+\node[squarednode] (commitcash)  [below=of contract]
+    {TxIn Alice 1000};
+\node[squarednode,align=center] (txOut1)       [right=of contract]
+    {Contract'\\value = 100\\dataScript = State \{\\commits = [Committed #1 Alice v:100 t:256]\}};
+\node[squarednode] (txOut2)       [below=of txOut1]
+    {Change AliceAddr 900};
+
+%Lines
+\draw[->,thick] (contract.east) -- (txOut1.west);
+\draw[->,thick] (commitcash.east) -- (txOut1.south);
+\draw[->,thick] (commitcash.east) -- (txOut2.west);
+\end{tikzpicture}
+
+
+\subsection{RedeemCC}
+
+Redeem a previously make CommitCash if valid.
+Alice committed 100 Ada with CC 1, timeout 256.
+
+\begin{tikzpicture}[
+squarednode/.style={rectangle, draw=orange!60, fill=orange!5, very thick, minimum size=10mm},
+]
+%Nodes
+\node[squarednode,align=center] (contract) {Contract\\redeemer = RC 1};
+\node[squarednode,align=center] (txOut1)       [right=of contract]
+    {Contract'\\dataScript = State \{commits = []\}};
+\node[squarednode] (txOut2)       [below=of txOut1]
+    {Change AliceAddr 900};
+
+%Lines
+\draw[->,thick] (contract.east) -- (txOut1.west);
+\draw[->,thick] (contract.east) -- (txOut2.west);
+\end{tikzpicture}
+
+
+\subsection{Pay}
+
+Alice pays 100 Ada to Bob.
+
+\begin{tikzpicture}[
+squarednode/.style={rectangle, draw=orange!60, fill=orange!5, very thick, minimum size=10mm},
+]
+%Nodes
+\node[squarednode,align=center] (contract) {Contract\\redeemer = Pay AliceSignature BobAddress v:100};
+\node[squarednode,align=center] (txOut1)       [right=of contract]
+    {Contract'\\dataScript = State \{commits - payment\}};
+\node[squarednode] (txOut2)       [below=of txOut1] {BobAddress 100};
+
+%Lines
+\draw[->,thick] (contract.east) -- (txOut1.west);
+\draw[->,thick] (contract.south) -- (txOut2.west);
+\end{tikzpicture}
+
+
+
+\section{Types and Data Representation}
+
+
+
+\begin{code}
+
+type Timeout = Int
+type Cash = Int
+
+type Person      = PubKey
+
+\end{code}
+
+\subsection{Identifiers}
+
+Commitments, choices and payments are all identified by identifiers.
+Their types are given here. In a more sophisticated model these would
+be generated automatically (and so uniquely); here we simply assume that
+they are unique.
+
+\begin{code}
+newtype IdentCC = IdentCC Int
+               deriving (Eq, Ord, Generic)
```

Comment:
You don't need the generic instances any more.

Reply:
Probably not for this PR, but would it be feasible to include the marlowe repo as a submodule and use the actual types from there? It would mess up your lhs exposition, I guess.

Reply:
The types are slightly different (Int/Integer) anyway, so as the implementation, so it's really hard to reuse things. I'll add links to Marlowe repo in Introduction and Bibliography.